app/models/course/settings/lesson_plan_component.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+class Course::Settings::LessonPlanComponent < Course::Settings::Component
+  include ActiveModel::Conversion
+
+  MILESTONES_EXPANDED_VALUES = %w(all none current).freeze
+
+  # Returns the setting which controls which milestones groups are expanded when
+  # the lesson plan page is first loaded.
+  #
+  # @return [String] A value in MILESTONES_EXPANDED_VALUES
+  delegate :milestones_expanded, to: :settings
+
+  # Sets which milestones groups are expanded when the lesson plan page is first loaded.
+  #
+  # @return [String] The new setting
+  def milestones_expanded=(setting)
+    raise ArgumentError, 'Invalid lesson plan milestone groups expanded setting.' \
```

Comment:
We should find some way to surface this `ArgumentError` in the controller to respond with the error message, so that the frontend can show this.

Reply:
Don't think it's necessary, the values come from the radio buttons. While it's all open source anyway, there's no need to be too helpful if anyone tries to fuzz the values.

Reply:
Even if we don't surface the error, it is probably good if we can catch ArgumentError, then return a `head :bad_request`. I'm not sure what happens if we raise this error but fail to catch it in the controller.

Reply:
A `500 Internal Server Error` response will be sent and the failure message will be shown. Since this should not be raised under any normal circumstance, we shouldn't handle it (or we should at least re-throw), but let it surface to the top so that rollbar etc can report the abnormal behavour to us?

Reply:
For errors that we know might occur, we should handle (and choose if we want to send that to Rollbar), and raise things that we do not expect to Rollbar. That gives us the ability to show relevant error messages if need be.

I think we can take that discussion separately as it is more of a convention. For the current state, I think it is fine 👍